services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -369,31 +378,46 @@ async def _observing_single_service(service_name: str) -> None:
                 #   dy-sidecar, dy-proxy, or containers) -> it cannot be removed safely
                 # 4. service started, and failed on closing -> it cannot be removed safely
 
-                failed_while_saving_state_and_outputs = (
-                    scheduler_data.dynamic_sidecar.service_removal_state.can_save
-                    and scheduler_data.dynamic_sidecar.were_containers_created
-                )
-                if failed_while_saving_state_and_outputs:
+                if (
+                    scheduler_data.dynamic_sidecar.wait_for_manual_intervention_after_error
+                ):
                     # use-cases: 3, 4
                     # Since user data is important and must be saved, take no further
                     # action and wait for manual intervention from support.
+
+                    # After manual intervention service can now be removed
+                    # from tracking.
+
+                    modulo_divisor = _get_30_second_modulo_divisor(
+                        dynamic_scheduler.DIRECTOR_V2_DYNAMIC_SCHEDULER_INTERVAL_SECONDS
+                    )
+                    if (
+                        # trigger every ~30 seconds to reduce pressure on
```

Comment:
why not check with a real timepoint instead of that modulo thing? that is anyway depending on DIRECTOR_V2_DYNAMIC_SCHEDULER_INTERVAL_SECONDS

Reply:
This appeared better to me. I've refactored it a bit to be more readable.